File: app/models/movie.rb
class Movie < ActiveRecord::Base
  RATINGS = [
    ["-", ""],
    [1, "1"],
    [2, "2"],
    [3, "3"],
    [4, "4"],
    [5, "5"]
  ]
  def self.search(search)
    where("title LIKE ? OR studio LIKE ? OR year LIKE ? OR rating LIKE ? OR genre LIKE ? OR cast_member LIKE ? OR director LIKE ? OR screen_writer LIKE ?", "%#{search}%", "%#{search}%", "%#{search}%", "%#{search}%", "%#{search}%", "%#{search}%", "%#{search}%", "%#{search}%")
  end

  validates :title, presence: true
  validates :studio, presence: true
  validates :year, presence: true
  validates :rating, presence: true
  validates :genre, presence: true
  validates :cast_member, presence: true
  validates :director, presence: true
  validates :screen_writer, presence: true

  has_many :reviews
  has_many :users, through: :reviews
end

File: react/src/components/Movie.js
import React from 'react'

let Movie = props => {
  let page = "";
  let movie_path = `movies/${props.id}`;
  if (props.showPage == props.id) {
    let reviews = props.reviewsData.map(review => {
      return(
        <li>
          <p>
            {review.title}
          </p>
          <p>
            {review.body}
          </p>
        </li>
      )
    })
    page =
    <div>
      <li>
        <p>{props.movieData.title}</p>
        <p>{props.movieData.studio}</p>
        <p>{props.movieData.year}</p>
        <p>{props.movieData.rating}</p>
        <p>{props.movieData.genre}</p>
        <p>{props.movieData.cast}</p>
        <p>{props.movieData.screen_writer}</p>
        <p>{props.movieData.director}</p>
        {reviews}
      </li>;
    </div>
  } else {
    page =
      <li onClick={props.onClick}>
        {props.data.title}
      </li>;
  }
  return(
    <div>
      {page}
    </div>

  )
}

export default Movie;
